minidump-processor/src/stackwalker/amd64.rs

```diff
@@ -0,0 +1,240 @@
+                Some(ctx.rbp)
+            } else {
+                None
+            }
+        }
+    };
+    // TODO: pointer-align this..? Does CALL push aligned ip values? Is sp aligned?
+    let last_sp = ctx.rsp;
+
+    let default_scan_range = 40;
+    let extended_scan_range = default_scan_range * 4;
+
+    // Breakpad gives more search range to the first frame of the stack for
+    // whatever reason, so let's duplicate that.
+    let scan_range = if let FrameTrust::Context = trust {
+        extended_scan_range
+    } else {
+        default_scan_range
+    };
+
+    for i in 0..scan_range {
+        let address_of_ip = last_sp + i * POINTER_WIDTH;
+        let caller_ip = stack_memory.get_memory_at_address(address_of_ip as u64)?;
+        if instruction_is_valid(caller_ip, modules) {
+            // ip is pushed by CALL, so sp is just address_of_ip + ptr
+            let caller_sp = address_of_ip + POINTER_WIDTH;
+
+            // Try to restore bp as well. This can be possible in two cases:
+            //
+            // 1. This function has the standard prologue that pushes bp and
+            //    sets bp = sp. If this is the case, then the current bp should be
+            //    immediately after (before in memory) address_of_ip.
+            //
+            // 2. This function does not use bp, and has just preserved it
+            //    from the caller. If this is the case, bp should be before
+            //    (after in memory) address_of_ip.
+            let mut caller_bp = None;
+
+            if let Some(last_bp) = last_bp {
+                let address_of_bp = address_of_ip - POINTER_WIDTH;
+                if last_bp == address_of_bp {
+                    let bp = stack_memory.get_memory_at_address(address_of_bp as u64)?;
+                    if bp > address_of_ip {
+                        caller_bp = Some(bp);
+                    }
+                } else if last_bp >= address_of_ip + POINTER_WIDTH {
+                    caller_bp = Some(last_bp);
+                }
+            }
+
+            let caller_ctx = CONTEXT_AMD64 {
+                rip: caller_ip,
+                rsp: caller_sp,
+                rbp: caller_bp.unwrap_or(0),
+                ..CONTEXT_AMD64::default()
+            };
+            let mut valid = HashSet::new();
+            valid.insert(INSTRUCTION_REGISTER);
+            valid.insert(STACK_POINTER_REGISTER);
+            if caller_bp.is_some() {
+                valid.insert(FRAME_POINTER_REGISTER);
+            }
+            let context = MinidumpContext {
+                raw: MinidumpRawContext::Amd64(caller_ctx),
+                valid: MinidumpContextValidity::Some(valid),
+            };
+            let mut frame = StackFrame::from_context(context, FrameTrust::Scan);
+            adjust_instruction(&mut frame, caller_ip);
+            return Some(frame);
+        }
+    }
+
+    None
+}
+
+fn instruction_is_valid(instruction: Pointer, modules: &MinidumpModuleList) -> bool {
```

Comment:
I know it's bikesheddy but I think I named the Breakpad function something like `InstructionAddressSeemsValid` because it's fundamentally just a heuristic and the universe is cruel.